src/Compilers/Core/Portable/Compilation/ConversionInfo.cs

```diff
@@ -0,0 +1,147 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Semantics;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// Summarizes whether a conversion is allowed, and if so, which kind of conversion (and in 
+    /// some cases, the associated symbol).
+    /// </summary>
+    public struct ConversionInfo : IEquatable<ConversionInfo>
+    {
+        private enum ConversionKind
```

Comment:
I didn't make it internal because no one external to this type needed it.  I just wanted to avoid using an int internally, though there's very little benefit  actually added since we really are just doing bitwise operations.  I'mon the fence about what to do with this enum.